content/ros/ROS Introduction/presentation/ROS_Introduction.tex

```diff
@@ -0,0 +1,902 @@
+\end{itemize}
+\end{frame}
+
+\subsection{Language independent}
+\begin{frame}{Features of ROS}
+\framesubtitle{Language independent}    
+\begin{itemize}
+    \item ROS functionalities are implemented as a library in different programming languages.
+    \item  These libraries are referred to as ROS client libraries.
+\end{itemize}
+\end{frame}
+
+\begin{frame}{Language independent}
+    \framesubtitle{Features of ROS}    
+
+    ROS client libraries.  
+    
+    \vspace{0.5cm}
+    \centering
+\scalebox{0.8}{%
+  \begin{minipage}{\textwidth}
+    \begin{itemize}
+        \item Main ROS Client libraries:
+        \begin{itemize}
+            \item roscpp \hspace{7cm} \includegraphics[width=.06\linewidth]{figures/cpp_logo.png}
+            \item rospy  \hspace{7.1cm} \includegraphics[width=.08\linewidth]{figures/python_logo.png}
+            \item roslisp \hspace{7cm} \includegraphics[width=.06\linewidth]{figures/lisp_logo.png}
+        \end{itemize}
+        \item Experimental ROS client libraries:
+        \begin{itemize}
+            \item rosjava \hspace{7cm} \includegraphics[width=.06\linewidth]{figures/java_logo.png}
+            \item rosruby \hspace{7cm} \includegraphics[width=.06\linewidth]{figures/ruby_logo.png}
+            \item and some others..
+        \end{itemize}
+        \item ROS support on MATLAB:
+        \begin{itemize}
+            \item Robotics System Toolbox \hspace{4.3cm} \includegraphics[width=.06\linewidth]{figures/matlab_logo.png}
+        \end{itemize}
+    \end{itemize}
+\end{minipage}%
+}
+\end{frame}
+
+
+\subsection{Distributed and Modular}
+
+\begin{frame}{Distributed and Modular}
+    \framesubtitle{Features of ROS}    
+    \begin{itemize}
+        \item ROS supports running processes on multiple computers connected together through a LAN.
+        \item In a system running ROS, there will be multiple of processes where each process can do
+        certain task. A process can be changed without altering the remaining processes.
+    \end{itemize}
+\end{frame}
+
+
+\subsection{A lot of libraries and tools}
+
+\begin{frame}{A lot of libraries and tools}
+    \framesubtitle{Features of ROS}    
+    \begin{itemize}
+        \item Examples of libraries:
+            \begin{itemize}
+                \item Navigation stack.
+                \item SLAM (gmapping, hector SLAM, etc..).
+                \item Localization (amcl, etc..).
+                \item Motion planning for manipulators (MoveIt)
+                \item Support for popular libraries (OpenCV, PCL).
+            \end{itemize}
+            
+        \item Examples of tools:
+            \begin{itemize}
+                \item RVIZ:3D Visualization.
+                \item ROS bag files: Logging Sensor Data.
+                \item Catkin: A Build System.
+                \item Command line tools.
+            \end{itemize}        
+    \end{itemize}
+\end{frame}
+
+
+\subsection{Bad Things About ROS}
+
+\begin{frame}{Bad Things About ROS}
```

Comment:
Well, you could add more here. For instance:
* it imposes a communication overhead in the name of modularity, even though careful software design could eliminate the need for communication in the first place. In fact, we could argue that ROS discourages careful thinking about software architectures in the first place
* in the Python world, it forces us to "live in the past" instead of using newer language versions (Python 2 is the default in Kinetic - although it is possible to make Python 3 work for the most part)

Reply:
Done. 
But what can I say when I mention this point ? What causes this overhead?
Regarding that ROS discourages careful thinking about software architectures, I think it's better to mention it during the presentation while discussing the communication overhead point, and not list it on the slide?

Reply:
The (potentially) unnecessary communication is what causes the overhead. Conceptual example: if we have two components X and Y - both written in the same language - we can call Y from X without using ROS if design the interface between the components sufficiently well. If we don't and we default to ROS, we'll have communication overhead.

The message that should come across here is that one should not approach every robotics software problem as if every problem is a nail and ROS is a hammer.

The point about the careful thinking can be said verbally, though it might be better to also have it on the slide for the benefit of future presenters.

Reply:
Ok I see.
>  The point about the careful thinking can be said verbally, though it might be better to also have it on the slide for the benefit of future presenters.

I might make a script text file that shows what to points are planned to be delivered during each slide.

Reply:
Yeah, that sound good.